applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py

```diff
@@ -0,0 +1,682 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+#
+# ==============================================================================
+
+# Making KratosMultiphysics backward compatible with python 2.6 and 2.7
+from __future__ import print_function, absolute_import, division
+
+# Kratos Core and Apps
+from KratosMultiphysics import *
+from KratosMultiphysics.ShapeOptimizationApplication import *
+
+# Additional imports
+from algorithm_base import OptimizationAlgorithm
+from custom_math import NormInf3D, Dot, ScalarVectorProduct, Norm2, RowSize, HorzCat, Minus, Plus, Trans, Prod, ZeroVector, ZeroMatrix, IsEmpty, SafeConvertVectorToMatrix
```

Comment:
in such a  case I would just import it as custom_math and then use it

Reply:
:+1: like one usually does with numpy to make it shorter
`import custom_math as cm`